pil_analyzer/src/pil_analyzer.rs

```diff
@@ -361,7 +370,10 @@ impl<T: FieldElement> PILAnalyzer<T> {
             id,
             source,
             absolute_name,
-            degree: self.polynomial_degree,
+            degree: self.polynomial_degree.unwrap_or_else(|| {
+                assert!(matches!(symbol_kind, SymbolKind::Constant()));
+                0
+            }),
```

Comment:
This is a bit weird and would require a deeper refactor: constants can be defined outside of namespaces, where the notion of degree does not exist. But we represent them as `Symbol` which requires passing a degree. 0 is already passed before this PR, I just made sure that we must be dealing a constant if this happens.

Reply:
Yes, degree should be removed from Symbol at some point.